File: src/main/java/com/alan/cockapooserver/blog/BlogService.java
package com.alan.cockapooserver.blog;

import lombok.RequiredArgsConstructor;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class BlogService {

    private final BlogRepository blogRepository;

    public List<Blog> getBlogList(){
        return blogRepository.findAll();
    }

    @Transactional
    public void updateBlogList() throws IOException {

        List<Blog> blogs = new ArrayList<>();

        String URL = "https://github.com/seongkyu-lim/TechBlogs";
        Document doc = Jsoup.connect(URL).get();
        Elements elements = doc.select("p[dir=auto]");

        for (int i=0; i<elements.size()-1; i+=2) {
            if(blogRepository.findByName(elements.get(i).text()).isPresent()) continue;
            Blog blog = new Blog();
            blog.setName(elements.get(i).text());
            blog.setUrl(elements.get(i+1).text());
            blogs.add(blog);
        }
        blogRepository.saveAll(blogs);
    }
}

File: src/main/kotlin/com/example/cockapoo/cockapoo_server/application/service/BlogService.kt

import org.jsoup.nodes.Document
import org.jsoup.select.Elements
import org.jsoup.Jsoup
import org.springframework.transaction.annotation.Transactional
import org.springframework.stereotype.Service

@Service
class BlogService(private val blogRepository: BlogRepository) {

    fun getBlogs(): List<Blog> = blogRepository.findAll()

    fun updateBlogsFromGithubRepository() {
        val blogs = _getBlogsFromGithubRepository()
        blogRepository.saveAll(blogs)
    }

    @Transactional
    private fun _getBlogsFromGithubRepository(): List<Blog> {
        val URL = "https://github.com/seongkyu-lim/TechBlogs"
        val doc: Document = Jsoup.connect(URL).get()
        val elements: Elements = doc.select("p[dir=auto]")

        return elements.chunked(2)
            .filter { chunk -> 
                chunk.size == 2 && blogRepository.findByTitle(chunk[0].text()) == null
            }
            .map { chunk -> 
                // Blog().apply {
                //     title = chunk[0].text()
                //     link = chunk[1].text()
                // }
                Blog(
                    title = chunk[0].text(),
                    link = chunk[1].text(),
                    content = "content",
                    author = "author"
                )
            }
    }
}